ChangeLog.d/pkparse_public_key_memleak.txt

```diff
@@ -0,0 +1,3 @@
+Bugfix
+   * Fix memory leak if mbedtls_pk_parse_public_key() call is repeated with the
```

Comment:
The documentation of `mbedtls_pk_parse_public_key` states that you must not do this. The change does fix another case which is an unambiguous bug, however: `mbedtls_pk_parse_public_key()` can leak memory in a low-memory condition (`pem_read_buffer` succeeds then the `alloc` function called by `pk_setup` fails to allocate memory). So please change the text here to mention only this bug.

If we decide that calling `mbedtls_pk_parse_public_key` on an already-set up context is allowed, we need to update the documentation and we need a unit test.